WcaOnRails/app/models/competition.rb

```diff
@@ -567,13 +587,25 @@ def has_base_entry_fee?
   end
 
   def has_fees?
-    base_entry_fee_lowest_denomination + competition_events.sum(:fee_lowest_denomination) > 0
+    if base_entry_fee_lowest_denomination.nil?
+      competition_events.sum(:fee_lowest_denomination) > 0
+    else
+      base_entry_fee_lowest_denomination + competition_events.sum(:fee_lowest_denomination) > 0
+    end
+  end
+
+  def entry_fee_required?
+    isConfirmed? && created_at.present? && created_at > Date.new(2018, 7, 11)
   end
 
   def competitor_limit_enabled?
     competitor_limit_enabled
   end
 
+  def automatic_registration_requirements?
```

Comment:
I'm concerned about having a method like this. As part of this PR, you've added new structured registration requirements A, B, and C. Consider a future where we want to introduce some new kind of structured registration information called D. Then, there would exist 3 kinds of competitions:

1. Sufficiently old competitions do not have to specify A, B, C, *nor* D, because these competitions were created before A, B, C, and D all existed.
2. Competitions that are required to have A, B, and C, but are *not* required to specify D, because they were created before D existed.
3. Brand new competitions that are required to specify A, B, C, *and* D.

In that world, I don't think this `automatic_registration_requirements?` function is very useful, because there is no longer one thing called "automatic registration requirements". Instead, there is a set of registration fields, which we introduced gradually over time. I think instead of this `automatic_registration_requirements?` method, it would be better to have one method per registration field called `foo_required?`. This is very similar to what you did in https://github.com/thewca/worldcubeassociation.org/pull/3086/files with `entry_fee_required?`.

Reply:
This makes total sense, I'll change this :)